extensions-contrib/kubernetes-overlord-extensions/src/main/java/org/apache/druid/k8s/overlord/KubernetesPeonLifecycle.java

```diff
@@ -203,12 +203,11 @@ protected synchronized TaskStatus join(long timeout) throws IllegalStateExceptio
     finally {
       try {
         saveLogs();
-        shutdown();
       }
       catch (Exception e) {
-        log.warn(e, "Cleanup failed for task [%s]", taskId);
+        log.warn(e, "Log processing failed for task [%s]", taskId);
       }
-
+      shutdown();
```

Comment:
Can `shutdown()` throw an exception? since it is making a call to the kubernetes client. If so, the stopTask should probably be in it's own finally block

Reply:
i decided to group saveLogs and shutdown together since they are both k8s lifecycle cleanup actions (it is okay if one fails), and then moved stopTask to a finally block b/c it has to happen